### PR TITLE
MyRadio_Podcast::getPodgetPodcastsAttachedToUser: fix API

### PR DESCRIPTION
**Pull request by markspolakovs**
The API thinks a MyRadio_User param is an int. It is not.

### CODE REVIEW

**Comment by markspolakovs** [comment]
Tested working on staging with the latest commit. To try it yourself, go to https://ury.org.uk/api-staging/v2/podcast/podcastsattachedtouser?api_key=YOUR_API_KEY&user=AN_USER_ID

**Comment by markspolakovs** [comment]
Wait, hang on, it's misbehaving.

**Comment by markspolakovs** [comment]
Right, that should work.

**Comment by markspolakovs** [comment]
r?'ing @mstratford as this now touches some MyRadio internals and I'd appreciate an extra pair of eyes

File: src/Classes/ServiceAPI/MyRadio_Podcast.php [hunk 1]
```diff
@@ -174,26 +174,32 @@ protected function __construct($podcast_id)
...
+     * @param int|string|MyRadio_User $user Default current user.
      *
      * @return int[]
      */
-    public static function getPodcastIDsAttachedToUser(MyRadio_User $user = null)
+    public static function getPodcastIDsAttachedToUser($user = null)
     {
         if ($user === null) {
             $user = MyRadio_User::getInstance();
+        } elseif (is_int($user)) {
```
Comment (mstratford) [inline src/Classes/ServiceAPI/MyRadio_Podcast.php:197]: May just be GitHub mobile, but is the indentation correct here?

**Comment by mstratford** [comment]
I've never been 100% sure how the mixins work, I'd appreciate finding out why the mixins come into these changes?

**Review by mstratford — COMMENTED** [review]
Seems generally fine, but why mixins?

**Comment by markspolakovs** [comment]
@mstratford Mixins are only ever used in `ServiceAPI::toDataSource()` to customise what fields you want (e.g. compare https://ury.org.uk/api/v2/user/currentuser and https://ury.org.uk/api/v2/user/currentuser?mixins=training,shows,personal_data,all_officerships).

They shouldn't be finding their way into any other methods, yet when I used this branch as of the commit-before-last with a mixin (e.g. https://ury.org.uk/api-staging/v2/podcast/podcastsattachedtouser?api_key=YOUR_API_KEY&user=AN_USER_ID&mixins=show) I got an error about the `$user` parameter - sticking in a var_dump revealed that it was passing in the value of the `mixins` field into the `$user` parameter. The simplest fix was to filter it out like we already do with `api_key`.

File: src/Classes/ServiceAPI/MyRadio_Podcast.php (hunk 1 above)
Comment (markspolakovs) [reply src/Classes/ServiceAPI/MyRadio_Podcast.php:197]: Looks fine to me

**Pull request merged**